src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -2388,35 +2442,34 @@ static inline JSC::EncodedJSValue createJSBufferFromJS(JSC::JSGlobalObject* lexi
     JSValue distinguishingArg = args.at(0);
     JSValue encodingArg = argsCount > 1 ? args.at(1) : JSValue();
     auto* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
-
     if (distinguishingArg.isAnyInt()) {
```

Comment:
this should use the check that only does this operation once instead of twice

Reply:
it does now